packages/commonwealth/client/scripts/views/modals/AuthModal/common/ModalBase/ModalBase.tsx

```diff
@@ -0,0 +1,316 @@
+import { ChainBase, WalletId, WalletSsoSource } from '@hicommonwealth/shared';
+import useWallets from 'client/scripts/hooks/useWallets';
+import app from 'client/scripts/state';
+import AuthButton from 'client/scripts/views/components/AuthButton';
+import {
+  AuthTypes,
+  AuthWallets,
+  EVMWallets,
+} from 'client/scripts/views/components/AuthButton/types';
+import {
+  CWTab,
+  CWTabsRow,
+} from 'client/scripts/views/components/component_kit/new_designs/CWTabs';
+import clsx from 'clsx';
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { CWIcon } from '../../../../components/component_kit/cw_icons/cw_icon';
+import { CWText } from '../../../../components/component_kit/cw_text';
+import {
+  CWModalBody,
+  CWModalFooter,
+} from '../../../../components/component_kit/new_designs/CWModal';
+import { ModalBaseProps, ModalBaseTabs } from '../../types';
+import { EVMWalletsSubModal } from './EVMWalletsSubModal';
+import { EmailForm } from './EmailForm';
+import { MobileWalletConfirmationSubModal } from './MobileWalletConfirmationSubModal';
+import './ModalBase.scss';
+
+const MODAL_COPY = {
+  'create-account': {
+    title: 'Create account',
+    description: `Common is built on web3 technology that utilizes wallets. \nHow would you like to sign up?`,
+    showExistingAccountSignInFooter: true,
+  },
+  'sign-in': {
+    title: 'Sign into Common',
+    description: '',
+    showExistingAccountSignInFooter: false,
+  },
+};
+
+const ModalBase = ({
+  onClose,
+  onSuccess,
+  layoutType,
+  hideDescription,
+  customBody,
+  showAuthenticationOptionsFor,
+  showWalletsFor,
+  bodyClassName,
+  onSignInClick,
+}: ModalBaseProps) => {
+  const copy = MODAL_COPY[layoutType];
+
+  const [activeTabIndex, setActiveTabIndex] = useState<number>(
+    showAuthenticationOptionsFor?.includes('sso') &&
+      showAuthenticationOptionsFor.length === 1
+      ? 1
+      : 0,
+  );
+  useEffect(() => {
+    setActiveTabIndex(
+      showAuthenticationOptionsFor?.includes('sso') &&
+        showAuthenticationOptionsFor.length === 1
+        ? 1
+        : 0,
+    );
+  }, [showAuthenticationOptionsFor]);
+
+  const [isEVMWalletsModalVisible, setIsEVMWalletsModalVisible] =
+    useState(false);
+  const [isAuthenticatingWithEmail, setIsAuthenticatingWithEmail] =
+    useState(false);
+
+  const handleClose = async () => {
+    setIsAuthenticatingWithEmail(false);
+    setIsEVMWalletsModalVisible(false);
+    isWalletConnectEnabled &&
+      (await onResetWalletConnect().catch(console.error));
```

Comment:
@burtonator, now we log the error if it comes

Reply:
I'm fine with this but I know someone is going to complain that you're not using logger :)

Reply:
would be great to use the logger 😄. Not sure if we built one?